config/v1alpha1/types_cluster_monitoring.go

```diff
@@ -101,3 +110,171 @@ const (
 	// UserDefinedNamespaceIsolated enables monitoring for user-defined projects with namespace-scoped tenancy. This ensures that metrics, alerts, and monitoring data are isolated at the namespace level.
 	UserDefinedNamespaceIsolated UserDefinedMode = "NamespaceIsolated"
 )
+
+// alertmanagerConfig provides configuration options for the default Alertmanager instance
+// that runs in the `openshift-monitoring` namespace. Use this configuration to control
+// whether the default Alertmanager is deployed, how it logs, and how its pods are scheduled.
+//
+// +union
+// +kubebuilder:validation:XValidation:rule="self.deploymentMode == 'Deployed' ? has(self.deployed) : (self.deploymentMode == ”) ? !has(self.deployed) : false",message="deployed must be set when deploymentMode is Deployed, and must be unset otherwise"
+type AlertmanagerConfig struct {
+	// deploymentMode determines whether the default Alertmanager instance should be deployed
+	// as part of the monitoring stack.
+	// Allowed values are Deployed and omitted.
+	// When set to Deployed, the Cluster Monitoring Operator
+	// ensures that an Alertmanager instance is created and managed in the `openshift-monitoring` namespace.
+	// Use this field if you want to explicitly opt in to running a platform-level Alertmanager.
+	// When omitted, this means no opinion and the platform is left to choose a default which is subject to change over time.
+	//
+	// +unionDiscriminator
+	// +kubebuilder:validation:Enum="";Deployed
+	// +optional
```

Comment:
Thinking through the user flow a bit more, having this optional means that something like:
```yaml
spec:
  alertmanagerConfig: {}
```
is valid.

I don't think that is correct and this field should be required but still allow the empty string value for the case where the parent field is totally omitted and this defaults to the empty string.

I don't recall if I previously said this should be optional, but if I did - my apologies.

Reply:
TBH not sure because we have changed things many times :S